### ISSUE TITLE
msisync_to_archive does not work for normal rods user anymore.

### ISSUE THREAD

**Issue by cookie33** [irods/contrib#8]
We did an put of a file using a normal rods user

```
09:35 irodstest2.storage.sara.nl:/home/robertv
robertv$ iput -R eudat rpmbuild/RPMS/noarch/irods-eudat-b2safe-dpm-client-1.0-0.noarch.rpm tokkie.rpm

09:35 irodstest2.storage.sara.nl:/home/robertv
robertv$ ils -l
/bob/home/robertv:
 ...
  robertv           0 eudat;eudatCache        16372 2015-12-03.09:35 & tokkie.rpm
```
There is a rule fired which replicates it to archive:
```
09:35 irodstest2.storage.sara.nl:/home/robertv
robertv$ iqstat -a
id     name
10437
                        #writeLine("serverLog","filePath: $filePath");
                        *CompoundRescName="eudat"
                        *CacheRescName   ="*CompoundRescName;eudatCache";
                        *ArchiveRescName ="*CompoundRescName;eudatPnfs";
                        writeLine("serverLog","Execute command to replicate (in resource *CompoundRescName) $objPath ($filePath) to *ArchiveRescName, because of put");
                        msisync_to_archive("*CacheRescName", $filePath, $objPath );
                |

```

This now fails if it is a normal rodsuser and NOT the rodsadmin.
```
Dec  3 09:37:12 pid:17810 NOTICE: writeLine: inString = Execute command to replicate (in resource eudat) /bob/home/robertv/tokkie.rpm (/var/lib/eudatCache/home/robertv/tokkie.rpm) to eudat;eudatPnfs, because of put
XXXX - last: eudatCache
XXXX - prev: eudat
XXXX - resolve
XXXX - success
XXXX - auto_repl :: off
Dec  3 09:37:12 pid:17810 NOTICE: rsDataObjRepl - Failed to replicate data object.
msisync_to_archive - fileModified failed [[-]   iRODS/server/drivers/src/fileDriver.cpp:723:fileModified :  status [CAT_INSUFFICIENT_PRIVILEGE_LEVEL]  errno [] -- message [fileModified - Failed to call modified interface.]
        [-]     libcompound.cpp:496:repl_object :  status [CAT_INSUFFICIENT_PRIVILEGE_LEVEL]  errno [] -- message [Failed to replicate the data object [/bob/home/robertv/tokkie.rpm] for operation [sync_object]]

] - [-830000]
Dec  3 09:37:12 pid:17810 ERROR: executeRuleAction Failed for msisync_to_archive status = -830000 CAT_INSUFFICIENT_PRIVILEGE_LEVEL
Dec  3 09:37:12 pid:17810 NOTICE: executeRuleBody: Microservice or Action msisync_to_archive Failed with status -830000
Dec  3 09:37:12 pid:17810 DEBUG: execMicroService3: error when executing microservice
line 6, col 3
                        msisync_to_archive("*CacheRescName", $filePath, $objPath );
                        ^

Dec  3 09:37:12 pid:17810 NOTICE: postProcRunRuleExec: exec of freq: 1h DOUBLE UNTIL SUCCESS OR 6 TIMES
Dec  3 09:37:12 pid:17810 NOTICE: modExeInfoForRepeat: rulId=10437,opStatus=-830000,nextRepeatStatus=4
Dec  3 09:37:12 pid:17810 NOTICE: Rule id 10437 set to run again at 1449135432 (frequency 2h DOUBLE UNTIL SUCCESS OR 5 TIMES. ORIGINAL TIMES=6 seconds)
Dec  3 09:37:12 pid:17810 NOTICE: Agent exiting with status = -830000
```

This works for the rodsadmin. Can this be fixed again?

In a previous version this worked.

Greetings,

Robert Verkerk

**Comment by cookie33** [comment]
One which works looks as follows:
```
Dec  3 09:47:12 pid:18044 NOTICE: writeLine: inString = Execute command to replicate (in resource eudat) /bob/home/rods/test.txt (/var/lib/eudatCache/home/rods/test.txt) to eudat;eudatPnfs, because of put
XXXX - last: eudatCache
XXXX - prev: eudat
XXXX - resolve
XXXX - success
XXXX - auto_repl :: off
Dec  3 09:47:13 pid:18046 NOTICE: execCmd:cmd/univMSSInterface.sh argv:mkdir '/var/lib/eudatPnfs/home/rods'
Dec  3 09:47:13 pid:18049 NOTICE: execCmd:cmd/univMSSInterface.sh argv:chmod '/var/lib/eudatPnfs/home/rods' 750
Dec  3 09:47:13 pid:18052 NOTICE: execCmd:cmd/univMSSInterface.sh argv:syncToArch /var/lib/eudatCache/home/rods/test.txt /var/lib/eudatPnfs/home/rods/test.txt
Dec  3 09:47:13 pid:18055 NOTICE: execCmd:cmd/univMSSInterface.sh argv:chmod '/var/lib/eudatPnfs/home/rods/test.txt' 600
Dec  3 09:47:13 pid:18058 NOTICE: execCmd:cmd/univMSSInterface.sh argv:stat '/var/lib/eudatPnfs/home/rods/test.txt' 
Dec  3 09:47:13 pid:18044 NOTICE: postProcRunRuleExec: exec of freq: 1h DOUBLE UNTIL SUCCESS OR 6 TIMES
Dec  3 09:47:13 pid:18044 NOTICE: modExeInfoForRepeat: rulId=10439,opStatus=0,nextRepeatStatus=4
Dec  3 09:47:13 pid:18044 NOTICE: Agent exiting with status = 0
```
Note the difference in `prev`.

Where did the `educate` in the case it went wrong from?

**Comment by jasoncoposky** [comment]
this has been fixed here: 3a8512f6cdecee08e5cb4b3b6b7ff06ef1e44ea8 please pull and rebuild to confirm.

**Comment by cookie33** [comment]
Tested with a rodsadmin and with a rodsuser.

Passed with flying colors.

Thanks
